### ISSUE TITLE
Add Buffer support to crypto.hmac

### ISSUE THREAD

**Issue by ThomasWeiss** [arangodb/arangodb#1986]
I'm trying to port the following code from node.js to a Foxx script:

```
var key = new Buffer(accesskey, "base64");
var hmac = crypto.createHmac("sha256", key);
hmac.update(inputvalue);
var sig = hmac.digest("base64");
```

Basically, the key I need to use (`accessKey`) is provided as a base64 string, and I need to encode the result of the operation as a base64 string as well.

I _may_ have solved the first step by doing

```
crypto.hmac(new Buffer(accessKey, 'base64').toString(), ...
```

but then I'm not sure about the actual encoding of the result, so I don't really know how to convert it in base64.

**Comment by jsteemann** [comment]
Should be fixed via 0de556165952f541fccc4fa53a704c0a602aee75.
This makes crypto.hmac accept a Buffer object as well.

**Comment by jsteemann** [comment]
Fixed in 3.1.
